pkg/daemon/daemon.go

```diff
@@ -253,9 +264,28 @@ func (dn *Daemon) addProfileConfig(socketPath string, configFile string, nodePro
 	return nil
 }
 
+// Add fifo scheduling if specified in nodeProfile
+func addScheduling(nodeProfile *ptpv1.PtpProfile, cmdLine string) string {
+	if nodeProfile.PtpSchedulingPolicy != nil && *nodeProfile.PtpSchedulingPolicy == "SCHED_FIFO" {
+		if nodeProfile.PtpSchedulingPriority == nil {
+			glog.Errorf("Priority must be set for SCHED_FIFO; using default scheduling.")
+			return cmdLine
+		}
+		priority := *nodeProfile.PtpSchedulingPriority
+		if priority < 1 || priority > 65 {
+			glog.Errorf("Invalid priority %d; using default scheduling.", priority)
+			return cmdLine
+		}
+		return fmt.Sprintf("/bin/chrt -f %d %s", priority, cmdLine)
+	}
```

Comment:
also there is no validation if `PtpSchedulingPriority != nil` but `PtpSchedulingPolicy == nil` no error of message to the user

Reply:
In my opinion setting PtpSchedulingPriority for a PtpSchedulingPolicy other than SCHED_FIFO should just be ignored.
That said, if it is necessary to block, it should be done in the ptp operator.

That said, I don't think we really need to handle the case of invalid input.  I think it is reasonable that if a user both disables the validation, and uses invalid input, then they need to enter correct input or deal with the consequences.  The reason we added the webhook was to handle this.

With the current implementation of linuxptp-daemon, we don't have good error handling, and don't handle ptp4l not starting.  I don't think it is appropriate to add another error path until we have proper error handling.  I'm open to improving error handling in ptp4l, but that's separate from this.
Also, from a user perspective, no harm will come if the system uses default scheduling; it just may not be as accurate as with fifo scheduling, but is still better than ptp4l not running at all.

Reply:
if this is the path you want to go it's fine from my side